server/src/main/java/org/elasticsearch/cluster/routing/RoutingTable.java

```diff
@@ -585,7 +584,12 @@ public RoutingTable build() {
             if (indicesRouting == null) {
                 throw new IllegalStateException("once build is called the builder cannot be reused");
             }
-            RoutingTable table = new RoutingTable(version, indicesRouting.build());
+            RoutingTable table;
+            if (version == 0 && indicesRouting.isEmpty()) {
+                table = EMPTY_ROUTING_TABLE;
```

Comment:
I am not sure I understand when this helps us, can you elaborate and/or comment?

Reply:
Hmm this just leaked in here from the other branch. It doesn't really give a big performance boost, it's more of a cleanup and I suppose saves a couple of cycles around this code in tests :) I can back it out if it's too noisy?

Reply:
+ε to revert

Reply:
reverted :)